app/controllers/JudgmentNeutralCitationController.scala

```diff
@@ -0,0 +1,142 @@
+package controllers
+
+import auth.TokenSecurity
+import configuration.{GraphQLConfiguration, KeycloakConfiguration}
+import controllers.util.ConsignmentProperty._
+import org.pac4j.play.scala.SecurityComponents
+import play.api.i18n.{I18nSupport, Lang, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Request}
+import services.{ConsignmentMetadataService, ConsignmentService}
+import uk.gov.nationalarchives.tdr.validation.schema.JsonSchemaDefinition.{BASE_SCHEMA, RELATIONSHIP_SCHEMA}
+import uk.gov.nationalarchives.tdr.validation.schema.ValidationError
+
+import java.net.URLEncoder
+import java.util.UUID
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class JudgmentNeutralCitationController @Inject() (
+    val controllerComponents: SecurityComponents,
+    val graphqlConfiguration: GraphQLConfiguration,
+    val keycloakConfiguration: KeycloakConfiguration,
+    val consignmentService: ConsignmentService,
+    val consignmentMetadataService: ConsignmentMetadataService,
+    val messages: MessagesApi
+)(implicit val ec: ExecutionContext)
+    extends TokenSecurity
+    with I18nSupport {
+
+  def addNCN(consignmentId: UUID): Action[AnyContent] = judgmentUserAndTypeAction(consignmentId) { implicit request: Request[AnyContent] =>
+    val ncnValue: Option[String] = request.getQueryString(NCN).filter(_.nonEmpty)
+    val noNcnSelected: Boolean = request.getQueryString(NO_NCN).exists(_.contains("no-ncn-select"))
+    val effectiveNoNcnSelected = noNcnSelected && ncnValue.isEmpty
+    // UI does not allow both NCN and no NCN to be selected, so only send judgmentReference if noNCN is selected
+    val judgmentReference: Option[String] = if (effectiveNoNcnSelected) request.getQueryString(JUDGMENT_REFERENCE).filter(_.nonEmpty) else None
+    consignmentService
+      .getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+      .map(reference => Ok(views.html.judgment.judgmentNeutralCitationNumber(consignmentId, reference, request.token.name, ncnValue, effectiveNoNcnSelected, judgmentReference)))
```

Comment:
When we refresh the page or click on the back button from the upload page, can we call the API to get the data and render them on the page?